source/user-manual/agents/agent-connection.rst

```diff
@@ -3,50 +3,164 @@
+         :class: output
+
+         status='connected'
+
+   .. group-tab:: Windows
+
+      .. code-block:: Powershell
+
+         > Select-String -Path C:\Program Files (x86)\ossec-agent\wazuh-agent.state -Pattern "^status"
+
+      .. code-block:: console
+         :class: output
+
+         wazuh-agent.state:7:status='connected'
+
+
+   .. group-tab:: macOS
+
+      .. code-block:: console
 
-  # /var/ossec/bin/agent_control -lc
+         # sudo grep ^status /Library/Ossec/var/run/wazuh-agentd.state
 
-You can also check to see if an agent is connected correctly by verifying if the TCP connection to the manager is established:
+      .. code-block:: console
+         :class: output
+
+         status='connected'
+
+Using the `agent_control` utility from the server
+-------------------------------------------------
+
+You can check the status of any agent remotely by using the :doc:`agent_control <../reference/tools/agent-control>` utility found with the Wazuh server. To get the status of an agent, run the following command replacing the ``-i`` parameter with your agent ID.
+
+.. code-block:: console
+
+   # /var/ossec/bin/agent_control -i <YOUR_AGENT_ID> | grep ^\s+Status
 
 .. code-block:: console
+   :class: output
+
+      Status:     Active
+
+-  ``Active``: The Wazuh manager received the `keepalive` signal from the agent.
+-  ``Disconnected``: The Wazuh manager didn't receive the `keepalive` signal during the last 10 minutes.
+
+Using the Wazuh API
+-------------------
+
+In addition, you can check the status of an agent requesting to the Wazuh API the `statistical information of an agent <https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.agent_controller.get_component_stats>`_.
+
+.. code-block:: none
+
+   GET /agents/<YOUR_AGENT_ID>/stats/agent
+
+.. code-block:: JSON
+   :emphasize-lines: 5
+
+   {
+     "data": {
+       "affected_items": [
+         {
+           "status": "connected",
+           "last_keepalive": "2022-08-16T20:36:27Z",
+           "last_ack": "2022-08-16T20:36:30Z",
+           "msg_count": 1441,
+           "msg_sent": 2326,
+           "msg_buffer": 0,
+           "buffer_enabled": true
+         }
+       ],
+       "total_affected_items": 1,
+       "total_failed_items": 0,
+       "failed_items": []
+     },
+     "message": "Statistical information for each agent was successfully read",
+     "error": 0
+   }
+
+Checking network communication
+------------------------------
+
+Agent communication with the manager requires outbound connectivity from agent to manager. It uses the port ``1514/TCP`` by default.
+
+If the agent isn't connected, it may possibly mean the enrollment wasn't succesful. Check the :doc:`/user-manual/agent-enrollment/index` section for details on this. You can also check if a TCP connection to the manager is established to verify if an agent can connect to it. The result should match the agent and manager IP addresses.
```

Comment:
```suggestion
If the agent isn't connected, it may possibly mean the enrollment wasn't successful. Check the :doc:`/user-manual/agent-enrollment/index` section for details on this. You can also check if a TCP connection to the manager is established to verify if an agent can connect to it. The result should match the agent and manager IP addresses.
```